src/domain/release-archive.ts

```diff
@@ -81,7 +82,26 @@ export class ReleaseArchive {
     return new ModuleFile(extractedModulePath);
   }
 
+  private isSupportedTarball(): boolean {
+    if (this._diskPath.endsWith(".tar.gz")) {
+      return true;
+    }
+    if (this._diskPath.endsWith(".tar.xz")) {
+      return true;
+    }
+    return false;
+  }
+
   private async extractReleaseTarball(extractDir: string): Promise<void> {
+    if (this._diskPath.endsWith(".tar.xz")) {
+      let reader = fs.createReadStream(this._diskPath);
+      let writer = tar.x({
```

Comment:
nit: Throughout the PR, can you use `const` instead of `let` where possible?

Reply:
Updated `release-archive.ts` to use `const`.

I think `xzdec.ts` was already using `const` for all the places that a variable was only assigned to once. If you'd strongly prefer that only `const` be used (and never `let`) then I could update it to use single-static-assignment style.